From 63fa06e84e0495fe56286ba00810158e253afdd9 Mon Sep 17 00:00:00 2001
From: Jun Zhang <jun.zhang@arm.com>
Date: Tue, 28 Nov 2023 17:25:54 +0800
Subject: [PATCH] rss/kronos: Add TFM_FWU_AGENT macro and misc changes for FWU

The `TFM_FWU_AGENT` is used to distinguish the FWU logic in
common code. Some related changes for FWU:

1.`PLATFORM_SERVICE_OUTPUT_BUFFER_SIZE`: from 64 to 256
2.`PS_NUM_ASSETS`: from 10 to 20

Upstream-Status: Pending
Signed-off-by: Jun Zhang <jun.zhang@arm.com>
---
 platform/ext/target/arm/rss/kronos/config.cmake        | 1 +
 platform/ext/target/arm/rss/kronos/config_tfm_target.h | 6 ++++++
 2 files changed, 7 insertions(+)

diff --git a/platform/ext/target/arm/rss/kronos/config.cmake b/platform/ext/target/arm/rss/kronos/config.cmake
index 4eeed9c3b..df9f83959 100644
--- a/platform/ext/target/arm/rss/kronos/config.cmake
+++ b/platform/ext/target/arm/rss/kronos/config.cmake
@@ -36,6 +36,7 @@ set(TFM_PARTITION_INTERNAL_TRUSTED_STORAGE    ON   CACHE BOOL   "Enable Internal
 set(TFM_PARTITION_CRYPTO                      ON   CACHE BOOL   "Enable Crypto partition")
 set(TFM_MBEDCRYPTO_PLATFORM_EXTRA_CONFIG_PATH  ""  CACHE PATH   "Config to append to standard Mbed Crypto config, used by platforms to cnfigure feature support")
 set(RSS_COMMS_MAX_CONCURRENT_REQ               4   CACHE STRING "Maximum number of concurrent connection requests to RSS")
+set(TFM_FWU_AGENT                              ON   CACHE BOOL   "Whether to enable FWU Agent or not")

 # Once all options are set, set common options as fallback
 include(${CMAKE_CURRENT_LIST_DIR}/../common/config.cmake)
diff --git a/platform/ext/target/arm/rss/kronos/config_tfm_target.h b/platform/ext/target/arm/rss/kronos/config_tfm_target.h
index b70aeda65..0c070b564 100644
--- a/platform/ext/target/arm/rss/kronos/config_tfm_target.h
+++ b/platform/ext/target/arm/rss/kronos/config_tfm_target.h
@@ -29,4 +29,10 @@
 #define ATTEST_TOKEN_PROFILE_ARM_CCA           1
 #endif

+#undef PS_NUM_ASSETS
+#define PS_NUM_ASSETS                          20
+
+#undef PLATFORM_SERVICE_OUTPUT_BUFFER_SIZE
+#define PLATFORM_SERVICE_OUTPUT_BUFFER_SIZE    256
+
 #endif /* __CONFIG_TFM_TARGET_H__ */
--
2.25.1

